apps/dashboard/src/components/step-preview-hover-card.tsx

```diff
@@ -0,0 +1,183 @@
+import { Editor } from '@maily-to/core';
+import {
+  blockquote,
+  bulletList,
+  button,
+  columns,
+  divider,
+  hardBreak,
+  heading1,
+  heading2,
+  heading3,
+  image,
+  orderedList,
+  section,
+  spacer,
+  text,
+} from '@maily-to/core/blocks';
+import {
+  ChannelTypeEnum,
+  ChatRenderOutput,
+  GeneratePreviewResponseDto,
+  PushRenderOutput,
+  StepTypeEnum,
+} from '@novu/shared';
+import {
+  InAppPreview,
+  InAppPreviewActions,
+  InAppPreviewBell,
+  InAppPreviewBody,
+  InAppPreviewHeader,
+  InAppPreviewNotification,
+  InAppPreviewNotificationContent,
+  InAppPreviewPrimaryAction,
+  InAppPreviewSubject,
+} from './workflow-editor/in-app-preview';
+import { ChatPreview } from './workflow-editor/steps/chat/chat-preview';
+import { EmailPreviewHeader, EmailPreviewSubject } from './workflow-editor/steps/email/email-preview';
+import { PushPreview } from './workflow-editor/steps/push/push-preview';
+import { SmsPhone } from './workflow-editor/steps/sms/sms-phone';
+
+export type StepType = StepTypeEnum;
+
+interface StepPreviewProps {
+  type: StepType;
+  controlValues?: any;
+}
+
+export function StepPreview({ type, controlValues }: StepPreviewProps) {
```

Comment:
adds the ability to provide a step preview on hover

Reply:
I like the idea, but it feels to me that it needs a bit of polishing, first thing that catches me is that the preview doesn't have any borders, so it merges with the other backgrounds. And maybe some title like "Preview" or smth.

The second thing is that some Maily editor popovers appear.

The third is that it should be shown when I hover over the step card, not just the text.

Fourth, should we show that in the workflow editor if we have a sidebar preview?

![Screenshot 2025-01-14 at 17 37 02](https://github.com/user-attachments/assets/568d5f80-c04e-4073-8bc5-a1049bd3bf1c)